content/languages/javascript/mocha.md

```diff
@@ -3,6 +3,36 @@ title: Mocha
 tags: [javascript, reference, testing]
 ---
 
+# Mocha
+
+## Basic Setup
+
+### Solution
+
+```javascript
+const add = (a, b) => a + b;
+```
+
+### Tests
+
+```javascript
+const {assert, config, expect} = require("chai");
+
+// don't truncate assertion diff output
+config.truncateThreshold = 0;
+
+// if applicable to the challenge, don't truncate printing deeply-nested objects
+require("util").inspect.defaultOptions.depth = null;
```

Comment:
Not sure if we should recommend adding this in tests. Does this affect the failure message?

Reply:
the `truncateThreshold` part? it's a must have. The vast majority of the kata are undebuggable without it.

Reply:
I think kaz is referring to `require("util").inspect.defaultOptions.depth = null;`. I've been using it at Qualified for awhile and haven't seen any issues, but that doesn't necessarily mean it isn't causing a problem somewhere.

The problem it solves is that logging deeply-nested objects is truncated:

```js
Welcome to Node.js v18.4.0.
Type ".help" for more information.
> console.log({a:{a:{a:{a:{a:{}}}}}})
{ a: { a: { a: [Object] } } }
undefined
> require("util").inspect.defaultOptions.depth = null;
null
> console.log({a:{a:{a:{a:{a:{}}}}}})
{
  a: {
    a: { a: { a: { a: {} } } }
  }
}
```

But this doesn't seem to affect Chai assertion failure messages, which truncate the depth either way. So it's mainly a convenience I add to challenges with deeply-nested object structures for users who might `console.log()` to debug. I can leave it out if that's the consensus--they can use `console.dir(foo, {depth: null})` or add it themselves.

Reply:
Ah, I didn't remember how the comments are added in between/around lines of code, sorry...

Reply:
No prob. Since it doesn't seem to affect the tests, I'll leave the suggestion out.